department-of-veterans-affairs/caseflow
# Queue | Add ability to sort by Days Waiting

Resolves #5781 

### Description
This implements sorting to `<Table/>`, adds typing to `AttorneyTaskTable`, adds a `LegacyAppeal` type, and splits existing types into `models` and `state`.

`<Table/>` will now support sorting any column by passing a `sortValue: (rowObj)` function which should return the value to compare. Example, displays CaseDetailsLink, sorts by veteran name attribute:
```javascript
{
  header: COPY.CASE_LIST_TABLE_VETERAN_NAME_COLUMN_TITLE,
  valueFunction: (task) => <CaseDetailsLink
    task={task}
    appeal={this.getAppealForTask(task)}
    disabled={!task.attributes.task_id} />,
  sortValue: (task) => this.getAppealForTask(task, 'veteran_full_name')
}
```

### Acceptance Criteria 

### Testing Plan
1. Go to `/queue` as an attorney (`BVASCASPER1`), confirm page loads correctly
2. Confirm Case Details, Docket Number, Issues, and Days Waiting columns are sortable

<details><summary>GIF</summary>
<img src="https://user-images.githubusercontent.com/8533004/42527722-4c60cfb4-842e-11e8-85eb-0ce5e1b560c1.gif">
</details>
